llvm/lib/SYCLLowerIR/ComputeModuleRuntimeInfo.cpp

```diff
@@ -28,13 +28,18 @@ constexpr int DebugModuleProps = 0;
 #endif
 
 namespace llvm::sycl {
+
 bool isModuleUsingAsan(const Module &M) {
-  NamedMDNode *MD = M.getNamedMetadata("device.sanitizer");
-  if (MD == nullptr)
-    return false;
-  assert(MD->getNumOperands() != 0);
-  auto *MDVal = cast<MDString>(MD->getOperand(0)->getOperand(0));
-  return MDVal->getString() == "asan";
+  for (const auto &F : M) {
+    if (F.getCallingConv() != CallingConv::SPIR_KERNEL)
+      continue;
+    if (F.arg_size() == 0)
```

Comment:
In which case is arg_size equal to 0? Isn't every kernel patched with __asan_launch arg in ExtendSpirKernelArgs?

In addition, why checking of __asan_launch arg is better than checking __AsanLaunchInfo like in the first commit?

Reply:
> In which case is arg_size equal to 0? Isn't every kernel patched with __asan_launch arg in ExtendSpirKernelArgs?

This function is used to check if ASan is enabled, so maybe "__asan_launch" isn't appended to kernel arguments (eg, single task kernel).

> In addition, why checking of __asan_launch arg is better than checking __AsanLaunchInfo like in the first commit?

A pure ESIMD device image doesn't have "__AsanLaunchInfo".

Reply:
> > In which case is arg_size equal to 0? Isn't every kernel patched with __asan_launch arg in ExtendSpirKernelArgs?
> 
> This function is used to check if ASan is enabled, so "__asan_launch" may not be appended to kernel arguments (eg, single task kernel).
> 

is single task kernel skipped in ExtendSpirKernelArgs?

Reply:
> > > In which case is arg_size equal to 0? Isn't every kernel patched with __asan_launch arg in ExtendSpirKernelArgs?
> > 
> > 
> > This function is used to check if ASan is enabled, so "__asan_launch" may not be appended to kernel arguments (eg, single task kernel).
> 
> is single task kernel skipped in ExtendSpirKernelArgs?

I means "isModuleUsingAsan" is also used when ASan is disabled.

Reply:
I see, thank you!